trie/patriciaMerkleTrie.go

```diff
@@ -713,6 +713,24 @@ func (tr *patriciaMerkleTrie) checkIfMigrationPossible(newVersion core.TrieNodeV
 	return nil
 }
 
+// IsMigratedToLatestVersion returns true if the trie is migrated to the latest version
+func (tr *patriciaMerkleTrie) IsMigratedToLatestVersion() (bool, error) {
+	tr.mutOperation.Lock()
+	defer tr.mutOperation.Unlock()
+
+	if check.IfNil(tr.root) {
+		return true, nil
+	}
+
+	version, err := tr.root.getVersion()
+	if err != nil {
+		return false, err
+	}
+
+	versionForNewlyAddedData := core.GetVersionForNewData(tr.enableEpochsHandler)
```

Comment:
To be researched if we can replace the enableEpochsHandler with the core.trieNodeVersionVerifier